src/material-experimental/mdc-chips/chip-grid.ts

```diff
@@ -364,17 +364,32 @@ export class MatChipGrid
    * @docs-private
    */
   setDescribedByIds(ids: string[]) {
+    // This gets called a lot, so avoid doing unnecessary work.
+    if (this._ariaDescribedbyIds === ids) {
```

Comment:
I think that instead of diffing the array and using the timeout, it would be much simpler to:
1. Remove the timeout altogether.
2. Change the implementation of `MatChipInput.setDescribedByIds` to be something like this:

```ts

setDescribedByIds(ids: string[]): void {
  const value = ids.join(' ');
  const element = this._elementRef.nativeElement;

  if (value) {
    element.setAttribute('aria-describedby', value);
  } else {
    element.removeAttribute('aria-describedby');
  }
}
```

Reply:
Yes, that's much better.
I wasn't familiar with this code and wasn't sure if there meant to be other implementations of the `MatChipTextControl` interface, so I just went for a local fix. Will send another PR.